lib/internal/fs/promises.js

```diff
@@ -37,10 +38,13 @@ const {
   validateUint32
 } = require('internal/validators');
 const pathModule = require('path');
+const { promisify } = require('internal/util');
 
 const kHandle = Symbol('handle');
 const { kUsePromises } = binding;
 
+const getDirectoryEntriesPromise = promisify(getDirectoryEntries);
```

Comment:
can you just abstract the native call? promisify isn't ideal...

Reply:
Can you clarify? Which native call? The only call `getDirectoryEntries` makes into binding code is when there's unknown types and so it has to call `stat`, which is why it's asynchronous (although it's synchronous using `statSync` when used by `readdirSync`), and used here as promisified.

I could make a promise-specific version of `getDirectoryEntries`, using the promise version of stat under the hood, but that seemed like overkill. (Maybe it's not?)